src/components/LaunchDetailsPage/LaunchDetailsPage.tsx

```diff
@@ -34,20 +55,20 @@ export const LaunchDetailsPage = () => {
           >
             <Grid item xs={4} sm={3} md={4}>
               <CardMedia
-                src={details.image}
-                alt={details.name}
+                src={launch.image}
+                alt={launch.name}
                 component="img"
                 className="card-media"
               />
             </Grid>
             <Grid item xs={4} sm={5} md={8} className="card-content">
-              <LaunchCardContent details={details} />
+              <LaunchCardContent launch={launch} />
             </Grid>
           </Grid>
         </CardActionArea>
       </Card>
     </Container>
   ) : (
-    <Navigate to={RoutesPath.HOME} />
+    <NotFoundPage />
```

Comment:
Why dont't you use Navigate to='*'?

Reply:
If it will be navigate we will loose url in search params, maybe if we go to link and we will see NotFoundPage on wrong id url params it can be better for user